File: src/Solutions/Euler285.hs
module Solutions.Euler285 where
import Text.Printf

expected1to10 = 10.20914 :: Double
f x = let u = sqrt (x - 1) in atan u * (x + 1) - atan (1 / u) * (x - 1) - 2 * u
expected n = let k = fromIntegral n in (f ((k + 1 / 2)^2) - f ((k - 1 / 2)^2)) / (2 * k)

euler285 :: IO String
euler285 = return $ printf "%.5f" $ (expected1to10+) $ sum $ map expected [11..100000]

File: rust/src/solutions/euler285.rs
fn f(x: f64) -> f64 {
    let u = (x - 1.0).sqrt();
    u.atan() * (x + 1.0) - (1.0 / u).atan() * (x - 1.0) - u - u
}

fn expected(n: u32) -> f64 {
    let k = n as f64;
    (f((k + 0.5).powi(2)) - f((k - 0.5).powi(2))) / (k + k)
}

pub fn euler285() -> String {
    let expected_1_to_10 = 10.20914;
    format!(
        "{:.5}",
        expected_1_to_10 + (11..=100000).map(expected).sum::<f64>()
    )
}
